src/Symfony/Component/Console/Formatter/OutputFormatter.php

```diff
@@ -134,21 +134,23 @@ public function format($message)
         $offset = 0;
         $output = '';
         $tagRegex = '[a-z][a-z0-9,_=;-]*+';
+        // Note: Offset capture is in bytes. For compatibility, explicitly use
+        // mbstring with ASCII encoding for string manipulations.
         preg_match_all("#<(($tagRegex) | /($tagRegex)?)>#ix", $message, $matches, PREG_OFFSET_CAPTURE);
         foreach ($matches[0] as $i => $match) {
             $pos = $match[1];
             $text = $match[0];
 
-            if (0 != $pos && '\\' == $message[$pos - 1]) {
+            if (0 != $pos && '\\' == mb_substr($message, $pos - 1, 1, 'ASCII')) {
                 continue;
             }
 
             // add the text up to the next tag
-            $output .= $this->applyCurrentStyle(substr($message, $offset, $pos - $offset));
+            $output .= $this->applyCurrentStyle(mb_substr($message, $offset, $pos - $offset, 'ASCII'));
```

Comment:
AFAIK, mb_substr also validates the charset, which means that this could fail if any non-ascii char is encountered
the proper charset to use is "8bit"

Reply:
I agree it should be `8bit` but the symfony mbstring polyfill component uses `ASCII` as the identifier for `8bit`.

Using ASCII lets the code work in Symfony for systems without the mbstring extension.

Reply:
still, using `substr_replace()` would be better IMHO :)

Reply:
This is an interesting idea. How do you suggest replacing a `substr` with a `substr_replace`?

Reply:
in this case it should be something like:
`substr_replace(substr_replace($message, '', 0, $offset), '', $pos - $offset)`
but I agree it's less readable...